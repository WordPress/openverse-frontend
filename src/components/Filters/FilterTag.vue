<template>
  <button
    class="filter-block button tiny tag mx-1"
    :aria-label="filterLabel + ' filter'"
  >
    <span>{{ filterLabel }}</span>
    <span
      class="close ml-2 p-2"
      tabindex="0"
      @click="onClick"
      @keyup.enter="onClick"
      ><i class="icon cross"
    /></span>
  </button>
</template>
<script>
export default {
  name: 'FilterTag',
  props: {
    code: String,
    filterType: String,
    label: String,
  },
  computed: {
    needsTranslation() {
      return !['audioProviders', 'imageProviders'].includes(
        this.$props.filterType
      )
    },
    filterLabel() {
      return this.needsTranslation
        ? this.$t(this.$props.label)
        : this.$props.label
    },
  },
  methods: {
    onClick() {
      this.$emit('filterChanged', {
        code: this.$props.code,
        filterType: this.$props.filterType,
      })
    },
  },
}
</script>
<style lang="scss" scoped>
.filter-block {
  margin-left: 0.5rem;
}
</style>
