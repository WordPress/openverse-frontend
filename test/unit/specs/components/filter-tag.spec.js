import FilterTag from '~/components/Filters/FilterTag'
import { render, screen } from '@testing-library/vue'
import userEvent from '@testing-library/user-event'

describe('FilterTag', () => {
  let options = {}
  let props = null
  beforeEach(() => {
    props = {
      code: 'foo',
      label: 'bar-foo',
      filterType: 'bar',
    }
    options = { propsData: props, listeners: { filterChanged: jest.fn() } }
  })

  it('should render correct contents', () => {
    const { container } = render(FilterTag, options)
    expect(container.firstChild).toMatchSnapshot()
  })

  it('should emit filterChanged event', () => {
    render(FilterTag, options)
    userEvent.click(screen.getByRole('button'))
    expect(options.listeners.filterChanged).toHaveBeenCalledWith({
      code: 'foo',
      filterType: 'bar',
    })
  })
})
