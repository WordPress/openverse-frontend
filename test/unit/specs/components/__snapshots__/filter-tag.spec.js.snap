// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FilterTag should render correct contents 1`] = `
<button
  aria-label="filters.aria.remove-filter"
  class="button tiny tag"
>
  <span>
    bar-foo
  </span>
   
  <span
    class="close ms-2 p-2"
  >
    <i
      class="icon cross"
    />
  </span>
</button>
`;
